### DISCUSSION TITLE
Can't Auto-Update Sitemap Using Nuxt SSR

### DISCUSSION THREAD

**Discussion by riyaz7us** [nuxt/nuxt#29119] [Questions] [answered]
Hi, I've been using Nuxt 3 SPA for months until I had to move to SSR. I created sitemap configuration and setup route rules in `nuxt.config.ts`. 
Based on the code , I expected Nuxt to update my sitemaps every 60 seconds but it only updates the sitemap during build. 
When I use `pnpm build` it builds and generates the correct sitemap during build, then it provides an instruction to start the server using `node ./.output/server/index.mjs` 
After I start the server, I expect it to update sitemap every 60 seconds (based on sitemap config) or every hour (based on routeRules), but nothing works. Is my assumption right about the sitemap being updated at every predefined interval when the SSR server starts?

Following is the routeRules and sitemap config in nuxt config:
```
  routeRules: {
    "/": { ssr: false },
    "/blog": { ssr: false },
//Tried this too but to no avail
    "/blog/**": { swr: true, sitemap: { changefreq: 'hourly', priority: 0.3 } },
    "/app": { ssr: false },
    "/app/**": { ssr: false },
    "/login": { ssr: false },
  },

site: {
    url: "https://example.com",
    name: "MyWebsite",
    description:
      "my website description",
    defaultLocale: "en",
    lang: "en",
  },

sitemap: {
    cacheMaxAgeSeconds: 60, // by this I expect sitemap to update every 30 seconds
    sitemaps: {
      pages: {
        includeAppSources: true,
        exclude: ["/app/**","/blog/**"],
      },
      posts: {
        includeAppSources: true,
        //Code below fetches blog posts from backend and generates an array of URLs from slugs
        urls: async function () {
          const posts = await fetch("https://example.com/api/blog").then((res) => res.json());
          return posts.map((p: Object) => `/blog/${p.slug}`);
        },
//I had to exclude these from posts sections because they're already in pages section
        exclude: ["/app/**", "/about-us", "/contact-us", "/privacy-policy", "/terms-of-service", "/refund-policy","/login","/pricing","/"],
        defaults: { priority: 0.7 },
      },
    },
  },

```

**Comment by TheAlexLichter** [comment, marked as answer]
Can you please open a discussion in the corresponding module repo https://github.com/nuxt-modules/sitemap and not in the core?

Also, the issue might be that you don't use a nitro API endpoint but a function inside your nuxt.config.ts which will only run at build time.
